pkg/processor/trigger/rabbitmq/trigger.go

```diff
@@ -105,114 +123,243 @@ func (rmq *rabbitMq) setEmptyParameters() {
-		true,                        // no-wait
-		nil,                         // args
+		rmq.configuration.QueueName,
+		rmq.consumerName,
+		false, /* auto-ack */
+		false, /* exclusive */
+		false, /* no-local */
+		true,  /* no-wait */
+		nil,   /* args */
 	)
 	if err != nil {
 		return errors.Wrap(err, "Failed to start consuming messages")
 	}
 
 	rmq.Logger.DebugWith("Starting consumption from queue", "queueName", rmq.configuration.QueueName)
+	return nil
+}
 
+func (rmq *rabbitMq) handleConnectionError(handleErr *amqp.Error) error {
+	rmq.Logger.WarnWith("Broker connection closed",
+		"err", handleErr.Error(),
+		"serverErr", handleErr.Server,
+		"recover", handleErr.Recover,
+		"reason", handleErr.Reason,
+		"code", handleErr.Code)
+
+	// best effort closing broker connection and channel
+	if err := rmq.brokerConn.Close(); err != nil {
+		rmq.Logger.WarnWith("Failed to close broker connection", "err", err.Error())
+	}
+	if err := rmq.brokerChannel.Close(); err != nil {
+		rmq.Logger.WarnWith("Failed to close broker channel", "err", err.Error())
+	}
+
+	// TODO: get durations from configuration
+	// try reconnecting for a while
+	if err := rmq.reconnect(10*time.Minute, 30*time.Second); err != nil {
+		return errors.Wrap(err, "Failed to reconnect to broker")
+	}
+
+	// start message consumption again
+	if err := rmq.consume(); err != nil {
+		return errors.Wrap(err, "Failed to start consuming messages")
+	}
 	return nil
 }
 
-func (rmq *rabbitMq) handleBrokerMessages() {
-	for message := range rmq.brokerInputMessagesChannel {
+func (rmq *rabbitMq) processMessage(message *amqp.Delivery) {
+
+	// bind to delivery
 
-		// bind to delivery
-		rmq.event.message = &message
+	// TODO: when moving to multiworkers - need to create event per message
+	rmq.event.message = message
+	rmq.event.SetID(nuclio.ID(message.MessageId))
 
-		// submit to worker
-		_, submitError, _ := rmq.AllocateWorkerAndSubmitEvent(&rmq.event, nil, 10*time.Second)
+	// submit to worker
+	_, submitError, _ := rmq.AllocateWorkerAndSubmitEvent(&rmq.event, nil, 10*time.Second)
+
+	// ack the message if we didn't fail to submit
+	if submitError == nil {
+		message.Ack(false) // nolint: errcheck
+	} else {
+		rmq.Logger.WarnWith("Failed to submit to worker", "err", submitError)
+	}
+}
+
+func (rmq *rabbitMq) getConsumerName() (string, error) {
+	var consumerName string
+	var err error
+	if common.IsInKubernetesCluster() {
 
-		// ack the message if we didn't fail to submit
-		if submitError == nil {
-			message.Ack(false) // nolint: errcheck
-		} else {
-			rmq.Logger.WarnWith("Failed to submit to worker", "err", submitError)
+		// in k8s, use the pod name and trigger id as the consumer name
+		consumerName = os.Getenv("HOSTNAME") + rmq.ID
+	} else if common.RunningInContainer() {
+		if consumerName, err = common.RunningContainerHostname(); err != nil {
+			return "", errors.Wrap(err, "Failed to get container hostname")
 		}
 	}
+	return consumerName, nil
```

Comment:
`consumerName` can potentially be empty?
how about setting it to some value when initialized?

Reply:
yes. will comment why it is ok